mmcv/runner/checkpoint.py

```diff
@@ -255,7 +255,7 @@ def load_from_local(filename, map_location):
     Returns:
         dict or OrderedDict: The loaded checkpoint.
     """
-
+    filename = osp.abspath(filename) if not osp.isabs(filename) else filename
```

Comment:
`isfile` can handle the relative path like `../../path/of/your/checkpoint` but not `~/path/of/your/file`. In addition, `abspath('~/path/of/your/file')` returns an unexpected path so we should use [os.path.expanduser](https://docs.python.org/zh-cn/3/library/os.path.html#os.path.expanduser). BTW, a unit test should be added to test those cases.